plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/JdbcIcebergClient.java

```diff
@@ -0,0 +1,246 @@
+        this.catalogId = requireNonNull(catalogId, "catalogId is null");
+    }
+
+    public List<String> getNamespaces()
+    {
+        ImmutableList.Builder<String> namespaces = ImmutableList.builder();
+        try (Handle handle = Jdbi.open(connectionFactory)) {
+            namespaces.addAll(handle.createQuery("" +
+                            "SELECT DISTINCT table_namespace " +
+                            "FROM iceberg_tables " +
+                            "WHERE catalog_name = :catalog")
+                    .bind("catalog", catalogId)
+                    .mapTo(String.class)
+                    .list());
+            namespaces.addAll(handle.createQuery("" +
+                            "SELECT DISTINCT namespace " +
+                            "FROM iceberg_namespace_properties " +
+                            "WHERE catalog_name = :catalog")
+                    .bind("catalog", catalogId)
+                    .mapTo(String.class)
+                    .list());
+        }
+        return namespaces.build();
+    }
+
+    public Map<String, Object> getNamespaceProperties(String namespace)
+    {
+        try (Handle handle = Jdbi.open(connectionFactory)) {
+            return handle.createQuery("" +
+                            "SELECT key, value " +
+                            "FROM iceberg_namespace_properties " +
+                            "WHERE catalog_name = :catalog AND namespace = :schema")
+                    .bind("catalog", catalogId)
+                    .bind("schema", namespace)
+                    .map((rs, ctx) -> new SimpleEntry<>(rs.getString("key"), rs.getString("value")))
+                    .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
+        }
+    }
+
+    public Optional<String> getNamespaceLocation(String namespace)
+    {
+        try (Handle handle = Jdbi.open(connectionFactory)) {
+            return handle.createQuery("" +
+                            "SELECT value " +
+                            "FROM iceberg_namespace_properties " +
+                            "WHERE catalog_name = :catalog AND namespace = :schema AND key = 'location'")
+                    .bind("catalog", catalogId)
+                    .bind("schema", namespace)
+                    .mapTo(String.class)
+                    .findOne();
+        }
+    }
+
+    public void createNamespace(String namespace, Map<String, Object> properties)
+    {
+        ImmutableList.Builder<NamespaceProperties> namespaceProperties = ImmutableList.builderWithExpectedSize(properties.size() + 1);
+        namespaceProperties.add(new NamespaceProperties(catalogId, namespace, "exists", "true"));
+        for (Entry<String, Object> property : properties.entrySet()) {
+            namespaceProperties.add(new NamespaceProperties(catalogId, namespace, property.getKey(), property.getValue().toString()));
+        }
+
+        try (Handle handle = Jdbi.open(connectionFactory)) {
```

Comment:
can `org.apache.iceberg.jdbc.JdbcCatalog::createNamespace` be used there instead of directly accessing the db table?